src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/WrapWithTag/WrapWithTagEndpoint.cs

```diff
@@ -70,6 +70,19 @@ public TextDocumentIdentifier GetTextDocumentIdentifier(WrapWithTagParams reques
         // Instead of C#, which certainly would be expected to go in an if statement, we'll see HTML, which obviously
         // is the better choice for this operation.
         var languageKind = _documentMappingService.GetLanguageKind(codeDocument, hostDocumentIndex, rightAssociative: true);
+
+        // However, reverse scenario is possible as well, when we have
+        // <div>
+        // |@if (true) {}
+        // <p></p>
+        // </div>
+        // in which case right-associative GetLanguageKind will return Razor and left-associative will return HTML
+        // We should hand that case as well, see https://github.com/dotnet/razor/issues/10819
+        if (languageKind is RazorLanguageKind.Razor)
+        {
+            languageKind = _documentMappingService.GetLanguageKind(codeDocument, hostDocumentIndex, rightAssociative: false);
+        }
+
         if (languageKind is not RazorLanguageKind.Html)
```

Comment:
I wonder if the logic in this `if` even gets hit any more, with the above change? Might be worth checking and possibly removing if it's redundant.

Reply:
Will check that and will add test(s) for the fix. Wanted to make sure it looked reasonable overall.

Reply:
Sounds good. Just maybe don't set auto-merge this time 😛

Reply:
The last ```if``` check is still needed for implicit C# expressions (see tests) - we might not be in either Razor or Html

Added and fixed tests. Most tests were partially wrong. After I made client connection mock Verifiable, two actually failed because they were verifying the opposite of what they should've.

Reply:
Looks like Dustin already fixed it though :) Merging his fix.